src/lib.rs

```diff
@@ -10,6 +10,8 @@ use ipc_gateway::{ApplyMsgParams, CrossMsg, StorableMsg};
 use num_derive::FromPrimitive;
 use num_traits::{FromPrimitive, Zero};
 
+pub use crate::atomic::{AtomicExecID, AtomicInput, AtomicInputID, AtomicOutput};
+pub use crate::atomic::{AtomicExecRegistry, AtomicInputState, LockableState};
```

Comment:
I wonder if we should make `atomic` module `pub`, also renaming it to `primitives`? Or event make it a separate create?

Reply:
Yep. What about maybe adding an `sdk` crate within this repo where we expose all the basic primitives to be used by actors supporting atomic_exec? In the future we can take this sdk and make it part of the `ipc_sdk` (@cryptoAtwill was going to kick-off the work on this soon).

Reply:
I was trying to understand better how Rust packaging system works and stumbled upon this:

> A package can contain as many binary crates as you like, but at most only one library crate. A package must contain at least one crate, whether that’s a library or binary crate.

It seems that we can't make this a separate library crate in this repository unless we convert the repo into a workspace. Have you meant converting this repo into a workspace?

Reply:
Or do you mean following the [root package](https://doc.rust-lang.org/cargo/reference/workspaces.html#root-package) approach?

Reply:
No strong preference, I leave it up to you. I was thinking to either add workspace, or even have it as a single library exposing several independent `mod` for namespacing.

Reply:
For reference, this is how we do it in `fvm_utils`: https://github.com/consensus-shipyard/fvm-utils/blob/main/Cargo.toml